/barlus/track_markers_node:
  ros__parameters:
    # The default RTSP stream address for the Barlus UW-S5-3PBX10 underwater camera.
    # If using MediaMTX, this should be the address of the MediaMTX server, e.g.,
    # stream_address: rtsp://192.168.1.17:8554/barlus
    stream_address: rtsp://192.168.1.10:554/user=admin&password=&channel=1&stream=0.sdp

    # The path to the camera info file. This file contains the camera calibration information.
    camera_info_url: package://barlus_image_pipeline/config/camera_info.yaml

    # The frame ID of the camera.
    frame_id: camera_link

    # The name of the camera.
    camera_name: barlus

    # The list of marker IDs to track. This should be updated according
    # to the markers used in the environment.
    marker_ids:
      - 0
      - 1

    # The size of the markers in meters.
    marker_size: 0.5

    # The markers' dictionary.
    # see https://github.com/opencv/opencv/blob/4ad4bd5dc09b343b2be6c8137b563f143d68584d/modules/objdetect/include/opencv2/objdetect/aruco_dictionary.hpp#L99
    dictionary: 0 # DICT_4X4_50

    # Whether or not to publish the TF of the detected markers.
    publish_tf: false
